numba/dppl/tests/dppl/test_atomic_op.py

```diff
@@ -87,6 +161,91 @@ def test_atomic_add1(self):
 
         self.assertTrue(np.all(ary == gold))
 
+
+    def test_atomic_sub_local_int32(self):
+        ary = np.random.randint(0, 32, size=32).astype(np.uint32)
+        orig = ary.copy()
+
+        dppl_atomic_sub = dppl.kernel('void(uint32[:])')(atomic_sub_int32)
+        with ocldrv.igpu_context(0) as device_env:
+            dppl_atomic_sub[32, dppl.DEFAULT_LOCAL_SIZE](ary)
```

Comment:
If local size is default why to set it explicitly? What if you will omit it?

Reply:
The way we are able to pass the global and local size is by using the __getitem__ attribute. Now, in the __getitem__() attribute, we receive these arguments using the parameter args (https://github.com/IntelPython/numba/blob/pydppl/numba/dppl/compiler.py#L304)

It behaves in the following way:

Case 1:
```
global_size = 32
dppl_atomic_sub[global_size]
```

If we test args in __getitem__() we will find it to be a single integer

Case 2:
```
global_size = (32, 32)
dppl_atomic_sub[global_size]
```
If we test args in __getitem__() we will find it to be a tuple of two integers (32, 32)

Case 3:
```
global_size = 32
local_size = 8
dppl_atomic_sub[global_size, local_size]
```
If we test args in __getitem__() we will find it to be a tuple of two integers (32, 8)

**Issue**: we will find identical type for both case 2 and 3 and therefore can not distinguish between which the user intended as global size and which as local size

Solution 1: Users always wrap whatever size they are passing in a list or tuple

So,

Case 1 becomes:
```
global_size = 32
dppl_atomic_sub[[global_size]] / dppl_atomic_sub[(global_size)] 
```

Users are always forced to wrap parameters as a list even for single value.

Solution 2: Always send two parameters:
Case 1 becomes:
```
global_size = 32
dppl_atomic_sub[global_size, some_default_value_if_user_don't_want_to_set_it]
```

I have chosen soultion 2. We can always discuss which is better or if we need to completely change how we allow users to specify the global and local size when invoking a kernel.

I hope this answers the question.